vertical-pod-autoscaler/pkg/utils/status/status_object_test.go

```diff
@@ -0,0 +1,254 @@
+			},
+			wantErr: false,
+		},
+		{
+			name: "getting status object - retryable error",
+			getReactor: func() func(action testcore.Action) (bool, runtime.Object, error) {
+				i := 0
+				return func(action testcore.Action) (bool, runtime.Object, error) {
+					if action.GetVerb() == "get" && action.GetResource().Resource == "leases" {
+						i++
+						switch i {
+						case 1, 2:
+							return true, nil, syscall.ECONNRESET
+						default:
+							return true, &apicoordinationv1.Lease{}, nil
+						}
+					}
+
+					return true, nil, fmt.Errorf("unsupported action")
+				}
+			}(),
+			wantErr: false,
+		},
+		{
+			name: "getting status object - non-retryable error",
+			getReactor: func() func(action testcore.Action) (bool, runtime.Object, error) {
+				i := 0
+				return func(action testcore.Action) (bool, runtime.Object, error) {
+					if action.GetVerb() == "get" && action.GetResource().Resource == "leases" {
+						i++
+						switch i {
+						case 1:
+							return true, nil, fmt.Errorf("non-retryable error")
+						default:
+							return true, &apicoordinationv1.Lease{}, nil
+						}
+					}
+
+					return true, nil, fmt.Errorf("unsupported action")
+				}
+			}(),
+			wantErr: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("test case: %s", tc.name), func(t *testing.T) {
+			fc := fake.NewSimpleClientset()
+			fc.PrependReactor("get", "leases", tc.getReactor)
+			client := NewClient(fc, leaseName, leaseNamespace, 10*time.Second, leaseName)
+			_, err := client.GetStatus()
+			assert.True(t, (err != nil) == tc.wantErr)
+		})
+	}
+}
+
+func TestIsStatusValid(t *testing.T) {
+	tests := []struct {
+		name         string
+		lease        *apicoordinationv1.Lease
+		leaseTimeout time.Duration
+		want         bool
+	}{
+		{
+			name: "Creation time check",
+			lease: &apicoordinationv1.Lease{
+				ObjectMeta: metav1.ObjectMeta{
+					CreationTimestamp: metav1.Time{Time: time.Now()},
+				},
+			},
+			leaseTimeout: 10 * time.Second,
+			want:         true,
+		},
+		{
+			name: "Renew time check",
+			lease: &apicoordinationv1.Lease{
+				ObjectMeta: metav1.ObjectMeta{
+					CreationTimestamp: metav1.Time{Time: time.Now().Add(-time.Minute)},
+				},
+				Spec: apicoordinationv1.LeaseSpec{
+					RenewTime: &metav1.MicroTime{Time: time.Now()},
+				},
+			},
+			leaseTimeout: 10 * time.Second,
+			want:         true,
+		},
+		{
+			name: "Outdated time check",
+			lease: &apicoordinationv1.Lease{
+				ObjectMeta: metav1.ObjectMeta{
+					CreationTimestamp: metav1.Time{Time: time.Now().Add(-time.Minute)},
+				},
+				Spec: apicoordinationv1.LeaseSpec{
+					RenewTime: &metav1.MicroTime{Time: time.Now().Add(-time.Minute)},
+				},
+			},
+			leaseTimeout: 10 * time.Second,
+			want:         false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("test case: %s", tc.name), func(t *testing.T) {
+			assert.Equal(t, IsStatusValid(tc.lease, tc.leaseTimeout), tc.want)
```

Comment:
Please add a message for failed equality check, otherwise we get "expected: false got: true" only on test failure

Reply:
I've updated test names to be more self-explanatory. Imho simple "expected, got" is enough.